pkg/cri/resource-manager/cache/cache.go

```diff
@@ -1110,6 +1119,37 @@ func (cch *cache) setEntry(key string, ptr, obj interface{}) error {
 	return nil
 }
 
+// checkFilePermissions check and warns if an existing file has unexpected permissions.
+func (cch *cache) checkFilePermissions(path string, perm os.FileMode, what string) {
+	if info, err := os.Stat(path); err == nil && !info.IsDir() {
+		expected := perm & os.ModePerm // deliberately ignore sticky bit in comparison
+		existing := info.Mode().Perm() & os.ModePerm
+		if existing != expected && (existing|expected) != expected {
+			cch.Warn("existing %s file %q has less strict permissions %v than expected %v",
```

Comment:
Why not error out?

Reply:
With a pre-created directory, the more liberal permissions might be a deliberate choice and we should be very careful when/if we try to outsmart the sysadmin. This is what we agreed in the last Fri session, but we should probably still error out in some extreme cases, for instance if a file or directory is unexpectedly world-readable. I see if I can update this to do that in some acceptable form...

Reply:
Added checking of permissions bits to reject.